### PR TITLE
Add logic to support BYOS version bump

### PR DESCRIPTION
**Pull request by chenwany**
Signed-off-by: chenwany <chenwany@amazon.com>

### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/setup.py
Comment (demartinofra) [inline cli/setup.py]: why is this needed?

File: util/bump-scheduler-plugin-version.sh
Comment (demartinofra) [inline util/bump-scheduler-plugin-version.sh]: Can we have this logic merged in the existing bump script?

File: cli/src/pcluster/schemas/cluster_schema.py
Comment (demartinofra) [inline cli/src/pcluster/schemas/cluster_schema.py]: 1.10 would not work. Also is the missing anchor at the beginning (`^`) a problem?

File: cli/setup.py
Comment (chenwany) [reply cli/setup.py]: Removed. It is not used

File: util/bump-scheduler-plugin-version.sh
Comment (chenwany) [reply util/bump-scheduler-plugin-version.sh]: Merged.
But with merging two script, we are putting it in the same step of the MCM. I was thinking about separating it like what we do for awsbatch cli. And in the MCM we can execute it as two steps.
After merge, I make the input of parallelcluster version as required, input of scheduler-plugin-version as optional since we don't update the plugin-version each time we update parallelcluster.
The benefit of merging the script is that we can do it in single commit. If we don't merge them, we are doing the version bump with two mcm steps with two PRs.

File: cli/src/pcluster/schemas/cluster_schema.py
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py]: Change to r"^[0-9]+\.[0-9]+$"

File: cli/src/pcluster/constants.py
Comment (lukeseawalker) [inline cli/src/pcluster/constants.py]: minor - are we ok with PLUGIN_ prefix or should it be SCHEDULER_PLUGIN?

File: cli/src/pcluster/validators/scheduler_plugin_validators.py
Comment (lukeseawalker) [inline cli/src/pcluster/validators/scheduler_plugin_validators.py]: would be easer if we define the supported version range as a constant?
initially, it would be
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["1.0",  "1.0"]
```
or, if it's easier
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_LOW_RANGE = "1.0"
SCHEDULER_PLUGIN_INTERFACE_VERSION_HIGH_RANGE = "1.0"
```

Then, we can easily compare with `plugin_version`
```
if not packaging.version.parse(SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[0]) <= packaging.version.parse(plugin_version) <= packaging.version.parse(SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[1]):
  error
```

Where for the error you can keep the same logic to build the more appropriate error string, e.g. SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[0] == SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[1] etc...

Comment (lukeseawalker) [reply cli/src/pcluster/validators/scheduler_plugin_validators.py]: Then, for version 2, we can decide to drop or keep 1.0 backward compatibility. e.g.
* drop case
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["2.0",  "2.0"]
```
* backward case
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["1.0",  "2.0"]
```

File: cli/tests/pcluster/schemas/test_cluster_schema.py
Comment (lukeseawalker) [inline cli/tests/pcluster/schemas/test_cluster_schema.py:615]: add `1.` and `.1` cases

File: util/bump-version.sh
Comment (lukeseawalker) [inline util/bump-version.sh]: nit - we can the _error_exit function

File: cli/src/pcluster/constants.py
Comment (chenwany) [reply cli/src/pcluster/constants.py]: Changed

File: cli/src/pcluster/validators/scheduler_plugin_validators.py
Comment (chenwany) [reply cli/src/pcluster/validators/scheduler_plugin_validators.py]: Changed

File: cli/tests/pcluster/schemas/test_cluster_schema.py
Comment (chenwany) [reply cli/tests/pcluster/schemas/test_cluster_schema.py:615]: Added

File: util/bump-version.sh
Comment (chenwany) [reply util/bump-version.sh]: Changed

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**
